src/main/app/src/pages/HomePage/KoulutuksetSection.tsx

```diff
@@ -75,61 +62,57 @@ const makeTableColumns = (t, organisaatioOid) => [
   {
     title: t('etusivu.kiinnitetytToteutukset'),
     key: 'toteutukset',
-    render: ({ toteutukset = 0 }) => (
-      <Badge color="primary">{toteutukset}</Badge>
+    render: ({ toteutusCount = 0 }) => (
+      <Badge color="primary">{toteutusCount}</Badge>
     ),
   },
 ];
 
-const KoulutuksetSection = ({ organisaatioOid, canCreate = true }) => {
+export const KoulutuksetSection = ({ organisaatioOid, canCreate = true }) => {
   const { t } = useTranslation();
 
-  const {
-    debouncedNimi,
-    showArchived,
-    page,
-    setPage,
-    orderBy,
-    setOrderBy,
-    tila,
-    filtersProps,
-  } = useFilterState({ paginationName: 'koulutukset' });
-
-  const watch = JSON.stringify([
-    page,
-    debouncedNimi,
-    organisaatioOid,
-    showArchived,
-    orderBy,
-    tila,
-  ]);
+  const filterState = useFilterState({ paginationName: 'koulutukset' });
 
-  const {
-    data: { result: koulutukset, pageCount = 0 } = {},
-    error,
-    reload,
-  } = useApiAsync({
-    promiseFn: getKoulutuksetFn,
-    nimi: debouncedNimi,
-    page,
-    showArchived,
-    organisaatioOid,
-    orderBy,
-    tila,
-    watch,
-  });
+  const { page, setPage, orderBy, setOrderBy, filtersProps } = filterState;
+
+  const params = useMemo(
+    () => getIndexParamsByFilters({ ...filterState, organisaatioOid }),
+    [filterState, organisaatioOid]
+  );
 
-  const rows = useMemo(() => {
-    return koulutukset
-      ? koulutukset.map(koulutus => ({ ...koulutus, key: koulutus.oid }))
-      : null;
-  }, [koulutukset]);
+  const {
+    data: { result: koulutukset, totalCount } = {},
+    isError,
+    isFetching,
+    refetch,
+  } = useSearchKoulutukset(params);
+
+  const [pageCount, setPageCount] = useState(0);
+  useEffect(() => {
+    if (totalCount !== undefined) {
+      setPageCount(Math.ceil(totalCount / FILTER_PAGE_SIZE));
+    }
+  }, [totalCount]);
+
+  const rows = useMemo(
+    () =>
+      koulutukset
+        ? koulutukset.map(koulutus => ({ ...koulutus, key: koulutus.oid }))
+        : null,
+    [koulutukset]
+  );
 
   const tableColumns = useMemo(() => makeTableColumns(t, organisaatioOid), [
     t,
     organisaatioOid,
   ]);
 
+  const hasRows = useMemo(() => !isFetching && !isError && koulutukset, [
```

Comment:
Varmaan voi käyttää react-queryn `success`-statusta tässä?
```suggestion
  const hasRows = useMemo(() => isSuccess && koulutukset, [
```